startup/30-geometry.py

```diff
@@ -1,45 +1,74 @@
-import time as ttime
+import numpy as np
+
 from ophyd import PseudoPositioner
-from ophyd import (PseudoSingle, EpicsMotor)
-from ophyd.pseudopos import (pseudo_position_argument, real_position_argument)
+from ophyd import PseudoSingle
+from ophyd import EpicsSignal
+import warnings
 
-too_small = 1.e-10
+# from ophyd import EpicsMotor
+from ophyd.pseudopos import pseudo_position_argument, real_position_argument
+from ophyd import Component as Cpt
+import bluesky.plans as bp
+import bluesky.plan_stubs as bps
+
+from ophyd.positioner import SoftPositioner
+
+
+class EpicsMotor(SoftPositioner):
+    def __init__(self, prefix, **kwargs):
+        super().__init__(**kwargs, init_pos=0)
+        self.prefix = prefix
 
-class Geometry(PseudoPositioner):
 
-    alpha = Cpt(PseudoSingle, '', labels=['geo'])
+class EpicsMotorWithLimits(EpicsMotor):
+    # low_limit = Cpt(EpicsSignal, ".LLM")
+    # high_limit = Cpt(EpicsSignal, ".HLM")
+    ...
 
-    th = Cpt(EpicsMotor, '{XtalDfl-Ax:Th}Mtr', labels=['geo'])
-    phi = Cpt(EpicsMotor, '{XtalDfl-Ax:Phi}Mtr', labels=['geo'])   
-#    phi = Cpt(EpicsMotor, '{XtalDfl-Ax:M7}Mtr', labels=['geo'])   
-    chi = Cpt(EpicsMotor, '{XtalDfl-Ax:Chi}Mtr', labels=['geo'])
-    tth = Cpt(EpicsMotor, '{XtalDfl-Ax:Tth}Mtr', labels=['geo'])
-    ih = Cpt(EpicsMotor, '{XtalDfl-Ax:IH}Mtr', labels=['geo'])
-    ir = Cpt(EpicsMotor, '{XtalDfl-Ax:IR}Mtr', labels=['geo'])    
+
+too_small = 1.0e-10
+
+
+class Geometry(PseudoPositioner):
+    # angles
+    alpha = Cpt(PseudoSingle, "")
+    beta = Cpt(PseudoSingle, "")
+    stth = Cpt(PseudoSingle, "")
+
+    # input motors
+    th = Cpt(EpicsMotor, "{XtalDfl-Ax:Th}Mtr", doc="Θ 3-circle theta for mono")
+    #
+    phi = Cpt(EpicsMotor, "{XtalDfl-Ax:Phi}Mtr", doc="Φ-stage sets bragg angle")
+    #    phi = Cpt(EpicsMotor, '{XtalDfl-Ax:M7}Mtr', labels=['geo'])
+    chi = Cpt(EpicsMotor, "{XtalDfl-Ax:Chi}Mtr", doc="Χ, beam steering")
+    tth = Cpt(EpicsMotor, "{XtalDfl-Ax:Tth}Mtr", doc="2Θ, spectrometer rotation")
+    ih = Cpt(EpicsMotor, "{XtalDfl-Ax:IH}Mtr", doc="input height")
+    ir = Cpt(EpicsMotorWithLimits, "{XtalDfl-Ax:IR}Mtr", doc="input rotation")
+
+    # Sample Motors
+    sh = Cpt(EpicsMotor, "bogus", doc="Sample vertical translation")
+    astth = Cpt(EpicsMotor, "bogus", doc="Sample rotation")
+    tblx2 = Cpt(EpicsMotor, "bogus", doc="Table 2 X")
+    # output motors
+    orb = Cpt(EpicsMotor, "bogus", doc="β, output arm rotation")
+    oh = Cpt(EpicsMotor, "bogus", doc="output arm vertical rotation")
```

Comment:
fill in the proper prefixes for these motors, remember they will be prefixed with `"XF:12ID1-ES"`